Compile error "redefinition of std::hash<QString> std::hash<QLatin1String>"

Compilation fails because of redefinition of `std::hash<QString>` and `std::hash<QLatin1String>`. I'm on Arch Linux using Qt 5.14.0. It seems like Qt added definition for these in [5.14](https://code.qt.io/cgit/qt/qtbase.git/commit/src/corelib/tools/qhashfunctions.h?h=5.14&id=4469e36d7203a55a4e158a50f0e9effc3f2fa3c2), which conflict with the ones in [src/backend/utils/HashMap.h](https://github.com/mmatyas/pegasus-frontend/blob/master/src/backend/utils/HashMap.h#L30).
```
cd src/ && ( test -e Makefile || /usr/bin/qmake -o Makefile /home/hagabaka/Code/pegasus-frontend/src/src.pro ) && make -f Makefile 
make[1]: Entering directory '/home/hagabaka/Code/pegasus-frontend/build/src'
cd backend/ && ( test -e Makefile || /usr/bin/qmake -o Makefile /home/hagabaka/Code/pegasus-frontend/src/backend/backend.pro ) && make -f Makefile 
make[2]: Entering directory '/home/hagabaka/Code/pegasus-frontend/build/src/backend'
g++ -c -pipe -O2 -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fPIC -std=gnu++11 -fno-rtti -fno-exceptions -Wall -Wextra -D_REENTRANT -DWITH_COMPAT_STEAM -DWITH_COMPAT_GOG -DWITH_COMPAT_ES2 -DWITH_COMPAT_LUTRIS -DWITH_COMPAT_SKRAPER -DQT_DEPRECATED_WARNINGS -DQT_DISABLE_DEPRECATED_BEFORE=0x050900 -DQT_NO_CAST_TO_ASCII -DQT_NO_PROCESS_COMBINED_ARGUMENT_START -DGIT_REVISION=\"alpha13-35-g5aa5521\" -DGIT_DATE=\"2019-12-16\" -DQT_NO_DYNAMIC_CAST -DQT_NO_EXCEPTIONS -DQT_NO_DEBUG -DQT_QUICK_LIB -DQT_GAMEPAD_LIB -DQT_GUI_LIB -DQT_QMLMODELS_LIB -DQT_QML_LIB -DQT_NETWORK_LIB -DQT_SQL_LIB -DQT_CORE_LIB -I/home/hagabaka/Code/pegasus-frontend/src/backend -I. -I/home/hagabaka/Code/pegasus-frontend/thirdparty -I/home/hagabaka/Code/pegasus-frontend/thirdparty/QtQmlTricks -I/home/hagabaka/Code/pegasus-frontend/thirdparty/SortFilterProxyModel -isystem /usr/include/qt -isystem /usr/include/qt/QtQuick -isystem /usr/include/qt/QtGamepad -isystem /usr/include/qt/QtGui -isystem /usr/include/qt/QtQmlModels -isystem /usr/include/qt/QtQml -isystem /usr/include/qt/QtNetwork -isystem /usr/include/qt/QtSql -isystem /usr/include/qt/QtCore -I. -I/usr/lib/qt/mkspecs/linux-g++ -o Api.o /home/hagabaka/Code/pegasus-frontend/src/backend/Api.cpp
In file included from /home/hagabaka/Code/pegasus-frontend/src/backend/modeldata/GameAssetsData.h:21,
                 from /home/hagabaka/Code/pegasus-frontend/src/backend/model/gaming/GameAssets.h:20,
                 from /home/hagabaka/Code/pegasus-frontend/src/backend/model/gaming/Game.h:20,
                 from /home/hagabaka/Code/pegasus-frontend/src/backend/model/gaming/Collection.h:20,
                 from /home/hagabaka/Code/pegasus-frontend/src/backend/Api.h:20,
                 from /home/hagabaka/Code/pegasus-frontend/src/backend/Api.cpp:18:
/home/hagabaka/Code/pegasus-frontend/src/backend/utils/HashMap.h:31:23: error: redefinition of ‘struct std::hash<QString>’
   31 |     template<> struct hash<QString> {
      |                       ^~~~~~~~~~~~~
In file included from /usr/include/qt/QtCore/qlist.h:47,
                 from /usr/include/qt/QtCore/qhash.h:46,
                 from /usr/include/qt/QtCore/QHash:1,
                 from /home/hagabaka/Code/pegasus-frontend/src/backend/utils/HashMap.h:20,
                 from /home/hagabaka/Code/pegasus-frontend/src/backend/modeldata/GameAssetsData.h:21,
                 from /home/hagabaka/Code/pegasus-frontend/src/backend/model/gaming/GameAssets.h:20,
                 from /home/hagabaka/Code/pegasus-frontend/src/backend/model/gaming/Game.h:20,
                 from /home/hagabaka/Code/pegasus-frontend/src/backend/model/gaming/Collection.h:20,
                 from /home/hagabaka/Code/pegasus-frontend/src/backend/Api.h:20,
                 from /home/hagabaka/Code/pegasus-frontend/src/backend/Api.cpp:18:
/usr/include/qt/QtCore/qhashfunctions.h:204:1: note: previous definition of ‘struct std::hash<QString>’
  204 | QT_SPECIALIZE_STD_HASH_TO_CALL_QHASH_BY_CREF(QString)
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/hagabaka/Code/pegasus-frontend/src/backend/modeldata/GameAssetsData.h:21,
                 from /home/hagabaka/Code/pegasus-frontend/src/backend/model/gaming/GameAssets.h:20,
                 from /home/hagabaka/Code/pegasus-frontend/src/backend/model/gaming/Game.h:20,
                 from /home/hagabaka/Code/pegasus-frontend/src/backend/model/gaming/Collection.h:20,
                 from /home/hagabaka/Code/pegasus-frontend/src/backend/Api.h:20,
                 from /home/hagabaka/Code/pegasus-frontend/src/backend/Api.cpp:18:
/home/hagabaka/Code/pegasus-frontend/src/backend/utils/HashMap.h:36:23: error: redefinition of ‘struct std::hash<QLatin1String>’
   36 |     template<> struct hash<QLatin1String> {
      |                       ^~~~~~~~~~~~~~~~~~~
In file included from /usr/include/qt/QtCore/qlist.h:47,
                 from /usr/include/qt/QtCore/qhash.h:46,
                 from /usr/include/qt/QtCore/QHash:1,
                 from /home/hagabaka/Code/pegasus-frontend/src/backend/utils/HashMap.h:20,
                 from /home/hagabaka/Code/pegasus-frontend/src/backend/modeldata/GameAssetsData.h:21,
                 from /home/hagabaka/Code/pegasus-frontend/src/backend/model/gaming/GameAssets.h:20,
                 from /home/hagabaka/Code/pegasus-frontend/src/backend/model/gaming/Game.h:20,
                 from /home/hagabaka/Code/pegasus-frontend/src/backend/model/gaming/Collection.h:20,
                 from /home/hagabaka/Code/pegasus-frontend/src/backend/Api.h:20,
                 from /home/hagabaka/Code/pegasus-frontend/src/backend/Api.cpp:18:
/usr/include/qt/QtCore/qhashfunctions.h:207:1: note: previous definition of ‘struct std::hash<QLatin1String>’
  207 | QT_SPECIALIZE_STD_HASH_TO_CALL_QHASH_BY_VALUE(QLatin1String)
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
make[2]: *** [Makefile:1993: Api.o] Error 1
make[2]: Leaving directory '/home/hagabaka/Code/pegasus-frontend/build/src/backend'
make[1]: *** [Makefile:75: sub-backend-make_first] Error 2
make[1]: Leaving directory '/home/hagabaka/Code/pegasus-frontend/build/src'
make: *** [Makefile:48: sub-src-make_first] Error 2
```

--- a/src/backend/utils/HashMap.h
+++ b/src/backend/utils/HashMap.h
@@ -26,6 +26,7 @@
 template <typename Key, typename Val, typename Hash = std::hash<Key>>
 using HashMap = std::unordered_map<Key, Val, Hash>;
 
+#if (QT_VERSION < QT_VERSION_CHECK(5, 14, 0))
 // hash for strings
 namespace std {
     template<> struct hash<QString> {
@@ -39,6 +40,7 @@ namespace std {
         }
     };
 }
+#endif
 
 // hash for enum classes
 struct EnumHash {

--- a/tests/backend/configfile/test_ConfigFile.cpp
+++ b/tests/backend/configfile/test_ConfigFile.cpp
@@ -119,7 +119,7 @@ void test_ConfigFile::merge_lines()
     QFETCH(QStringList, parts);
     QFETCH(QString, expected);
 
-    const std::vector<QString> parts_vec = parts.toVector().toStdVector();
+    const std::vector<QString> parts_vec(parts.cbegin(), parts.cend());
 
     QCOMPARE(metafile::merge_lines(parts_vec), expected);
 }

